main.cpp

```diff
@@ -54,51 +54,89 @@ Purpose:  This project will show you the difference between member functions and
 
 #include <iostream>
 #include <string>
+
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
-    //2
-    //3
+    T(int v, const char* name);   //1
+
+    int value; //2
+    std::string name;//3
 };
 
-struct <#structName1#>                                //4
+T::T(int v, const char* constructorName) :
+value(v),
+name(constructorName)
+{}
+
+struct CompareObjects                                //4
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    T* compare(T* a, T* b) //5
     {
+        if ( a == nullptr || b == nullptr)
+        {
+            std::cout << "Your pointer is not valid!" << std::endl;
+            return nullptr;
+        }
         if( a->value < b->value ) return a;
         if( a->value > b->value ) return b;
         return nullptr;
```

Comment:
Revise your logic so the `return nullptr;` statement only happens once in this function
- [ ] revise